### PR TITLE
fix(store): survive iframe/tag load races for direct-iframe embeds

### PR DESCRIPTION
**Pull request by Redskull-127**
## Problem

When a Surface form is embedded as a **direct iframe** (customer pastes the iframe, tag loads separately), the handshake was one-shot and answer-only:

- The tag deferred its `message` listener until DOMContentLoaded and never pushed data proactively, so a fast-booting iframe's `SEND_DATA` was silently dropped.
- Result: no attribution (UTMs/referrer/cookies), no journey stitching, duplicate-lead risk, no partial-fill — all degrading invisibly while the form still renders and submits.

## Changes

- **`src/store/message-listener.ts`** — attach the `window` message listener immediately. Nothing in the handler needs the DOM (iframes are queried at message time); the DOMContentLoaded deferral was pure message loss.
- **`src/store/store.ts`** — push `STORE_UPDATE` once on store init (at DOM-ready when loaded in `<head>`, immediately otherwise), plus `LEAD_DATA_UPDATE` only when cached lead data exists. Identify stays `SEND_DATA`-driven, so the tag still never creates a lead on pages where no form asked.
- **`test/direct-iframe.html`** (new) — reproduces the race: hard-coded iframe with a delayed, cache-busted tag loader (`?tagDelay=<ms>`, `?noTag=1`), SPA route-change button for the clobber scenario. The existing test pages create the iframe *via* the tag, so they can't reproduce this.

## Verification (browser, real deployed forms)

- **Race reproduced pre-fix mechanics:** form's `SEND_DATA` at ~3s unanswered when the tag was injected at 4.5s.
- **Rescue verified:** the init push delivered the payload to the already-running form; lead identity converged to the existing lead (single deduped identify, no duplicate).
- **Clobber check:** typed input survived a host `pushState` → route-change `STORE_UPDATE`.
- `pnpm run typecheck` + `pnpm run build` clean; bundle artifacts rebuilt.

Pairs with trysurface/surface_forms#4959 (forms-side: 500ms `SEND_DATA` polling + wider no-tag fallback + prefill guard). Each side is independently safe against old deployed versions of the other; deploy the forms side first (instant), tag second (CDN-cached).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/store/store.ts [hunk 1]
```diff
@@ -61,6 +61,20 @@ export class SurfaceStore {
       );
       this.setupRouteChangeDetection();
     }
+
+    // Direct-iframe embeds may have posted SEND_DATA before this script was
+    // listening; push once so they don't depend on that request being heard.
+    // LEAD_DATA_UPDATE only from cache — identify stays SEND_DATA-driven so the
+    // tag never creates a lead on pages where no form asked for one.
+    const pushInitialData = () => {
+      this.sendPayloadToIframes("STORE_UPDATE");
```
Comment (AminHakem) [inline src/store/store.ts:79]: issue(blocking): This fallback only replays synchronous store state. On a first visit, `getLeadDataWithTTL()` is empty, so the missed `SEND_DATA` never triggers `identifyLead`; `initializeUserJourneyTracking()` also starts its fetch without awaiting it, so this `STORE_UPDATE` is sent with `surfaceLeadData: null` and `userJourneyId: null`, and neither result is pushed later. I reproduced this with `?tagDelay=15000` and cleared storage: the iframe sent two `SEND_DATA` messages before tag load, no identify request ran, and the journey ID appeared only after this one-time push. That means compatibility with one-shot/old forms still loses lead identity and journey attribution. Please make missed-request recovery drive the full async path when a Surface iframe is present, then send updates after identity/journey state resolves.

File: test/direct-iframe.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        if (params.has("noTag")) return;
+        const delay = Number(params.get("tagDelay") ?? 0);
+        setTimeout(() => {
+          const s = document.createElement("script");
+          s.src = "../surface_tag.js?t=" + Date.now();
+          s.setAttribute(
+            "site-id",
+            params.get("siteId") || "cllo318mt0003mb08hnp0f5zy"
+          );
+          s.onload = () => window.hookTagSentEvents && window.hookTagSentEvents();
```
Comment (AminHakem) [inline test/direct-iframe.html]: issue(blocking): `s.onload` runs after the tag bundle has fully executed. `SurfaceStore` calls `pushInitialData()` synchronously during construction, before `window.SurfaceTagStore` is assigned and before this callback installs the wrapper, so the central fallback event is invisible to the new regression page. In the same 15-second reproduction, the page recorded two pre-tag `SEND_DATA` events but zero `STORE_UPDATE` events after tag load even though the fallback ran. Please instrument before tag execution or add a deterministic test around the `postMessage` boundary so this page can actually observe the behavior it is intended to cover.

**Review by AminHakem — COMMENTED** [review]
Two blocking race-handling gaps remain. Inline comments include the reproduction evidence and suggested direction.

File: test/direct-iframe.html (hunk 2 above)
Comment (AminHakem) [reply test/direct-iframe.html]: @greptile confirm

File: src/store/store.ts (hunk 1 above)
Comment (AminHakem) [reply src/store/store.ts:79]: @greptile confirm

File: test/direct-iframe.html (hunk 2 above)
Comment (Redskull-127) [reply test/direct-iframe.html]: yea i was able to reproduce this behaviour - fixing

File: src/store/store.ts (hunk 1 above)
Comment (Redskull-127) [reply src/store/store.ts:79]: good catch, this was real — the push only replayed whatever was synchronously available, so first-visit + one-shot forms got null lead/journey and nothing after. reworked in 18dd6a3:

- a surface iframe already in the DOM is now treated as the missed request itself — the init push runs `identifyLead` and follows up with `LEAD_DATA_UPDATE`, exactly like an answered `SEND_DATA` would. pages without a surface iframe stay quiet, so we still never create leads where no form asked.
- the journey id setter now re-pushes `STORE_UPDATE` when the id resolves, so iframes that already got the first push get the pageview stitched (this also fixes the same staleness on the SPA route-change path, which pushed before the journey update resolved).

re-ran your repro (cleared storage + long tagDelay against the deployed one-shot form): two pre-tag SEND_DATAs lost, then at tag load — STORE_UPDATE (journey null) → STORE_UPDATE (journey resolved) → identify from the tag → LEAD_DATA_UPDATE carrying both leadId and journeyId. one small note on "loses lead identity": the form self-identifies via its 100ms/1.5s fallback with an identical fingerprint algo, so identity converged server-side even before this fix — but the journey id was genuinely lost, and now both are delivered explicitly.

File: test/direct-iframe.html (hunk 2 above)
Comment (Redskull-127) [reply test/direct-iframe.html]: yep — the constructor pushed synchronously before index.ts assigned `window.SurfaceTagStore`, so the wrapper never saw the one event this page exists to observe. fixed in 18dd6a3 from both sides:

- the tag now defers the immediate-push branch by one task (`setTimeout(0)`), so the store is exposed on `window` before the push fires (the DOMContentLoaded branch already had this property)
- the test page installs a `defineProperty` setter trap on `window.SurfaceTagStore` and hooks `notifyIframe` at the moment of assignment, instead of waiting for script `onload`

with your 15s repro the init STORE_UPDATE, the journey-resolve STORE_UPDATE and the LEAD_DATA_UPDATE all show up in the event log now.

**Review by AminHakem — APPROVED** [review]
